### PR TITLE
test: add test to ensure immutable contracts cannot be changed

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**
Increase test coverage for `ContractUpdate`

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java [hunk 1]
```diff
@@ -601,4 +602,29 @@ final Stream<DynamicTest> playGame() {
                 .when(when.toArray(HapiSpecOperation[]::new))
                 .then(then.toArray(HapiSpecOperation[]::new));
     }
+
+    @HapiTest
+    final Stream<DynamicTest> cannotUpdateImmutableContractExceptExipry() {
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java:607]: typo: `Exipry` -> `Expiry`

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java [hunk 2]
```diff
@@ -601,4 +602,29 @@ final Stream<DynamicTest> playGame() {
...
+                newKeyNamed(ADMIN_KEY),
+                cryptoCreate(AUTO_RENEW_ACCOUNT),
+                cryptoCreate(STAKED_ACCOUNT),
+                uploadInitCode(CONTRACT),
+                contractCreate(CONTRACT).immutable(),
+                contractUpdate(CONTRACT).newAutoRenew(1).hasKnownStatus(MODIFYING_IMMUTABLE_CONTRACT),
+                contractUpdate(CONTRACT)
+                        .newAutoRenewAccount(AUTO_RENEW_ACCOUNT)
+                        .hasKnownStatus(MODIFYING_IMMUTABLE_CONTRACT),
+                contractUpdate(CONTRACT).newDeclinedReward(true).hasKnownStatus(MODIFYING_IMMUTABLE_CONTRACT),
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java:619]: Maybe it's just github but the indentation looks wrong

**Review by david-bakin-sl — DISMISSED** [review]
LGTM

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @lukelee-sl !

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Pull request merged**